src/App.tsx

```diff
@@ -20,17 +25,112 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+    step: 2,
+    frameSize: 2,
+    itemWidth: 130,
+    animationDuration: 1000,
+    infinite: false,
+  };
+
+  handleItemWidth = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({ itemWidth: Number(event.target.value) });
+  };
+
+  handleFrameSize = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({ frameSize: Number(event.target.value) });
+  };
+
+  handleStep = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({ step: Number(event.target.value) });
+  };
+
+  handleAnimationDuration = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({ animationDuration: Number(event.target.value) });
+  };
+
+  handleInfinite = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({ infinite: event.target.checked });
   };
 
   render() {
-    const { images } = this.state;
+    const {
+      images,
+      step,
+      frameSize,
+      itemWidth,
+      animationDuration,
+      infinite,
+    } = this.state;
 
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title">Carousel with {images.length} images</h1>
+        <div className="controls">
+          <label>
+            {'ItemWidth: '}
+            <input
+              type="number"
+              name="itemWidth"
+              id="ItemWidth"
+              defaultValue="130"
```

Comment:
```suggestion
              value={itemWidth}
```
Better use value from state, cause you have onChange handler that changes that value in state. So it will redraw it.